tests/unit/mixins/scoped-query-cache-test.js

```diff
@@ -139,3 +139,34 @@ test('if shouldCachePredicate is false, do not insert into cache', function(asse
 
 // test('findRecord is supported');
 // test('failed promise is handled properly')
+
+test('ScopedQueryCacheMixin#scheduleCacheDecay schedules the decay task', function(assert) {
+  let done = assert.async();
+  let ScopedQueryCacheObject = Ember.Object.extend(ScopedQueryCacheMixin);
+  let subject = ScopedQueryCacheObject.create();
+
+  subject.cache['key'] = 10;
+
+  subject.scheduleCacheDecay(50, () => {
+    assert.equal(Object.keys(subject.cache).length, 0);
+    done();
+  });
+});
+
+test('ScopedQueryCacheMixin#cancelDecay correctly cancels the cache deletion', function(assert) {
+  let done = assert.async();
+  let ScopedQueryCacheObject = Ember.Object.extend(ScopedQueryCacheMixin);
+  let subject = ScopedQueryCacheObject.create();
+
+  subject.cache['key'] = 10;
+
+  subject.scheduleCacheDecay(3000);
+  subject.cancelCacheDecay();
+
+  setTimeout(() => {
```

Comment:
Sinon has fake clock built in to simulate `setTimeout`, wonder if we can use that here